### PR TITLE
Add dentistry header changes

### PR DESCRIPTION
**Pull request by RitwikSrivastava**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #<gh-issue-id>

Test URLs:
- Original: https://www.sunstar-foundation.org/dentistry/children
- Before: https://main--sunstar-foundation--hlxsites.hlx.page/dentistry/children
- After: https://dentistry-header--sunstar-foundation--hlxsites.hlx.page/dentistry/children 

- [ ] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation
------------- | -------------
 For e.g. cards | [doc-link](https://main--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=0)

- [ ] New variations introduced in this PR

Variation Name    | Documentation
------------- | -------------
 For e.g. cards (grid)  | [doc-link](https://main--sunstar-foundation--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/sidekick/blocks/cards&index=2)

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -170,6 +177,17 @@ header nav.nav-bottom .menu-level-1-item > a {
   justify-content: space-between;
 }
 
+header nav.nav-middle h5 > a:hover {
+  line-height: 1;
+  text-decoration: none;
+  background-color: var(--secondary);
```
Comment (jindaliiita) [inline blocks/header/header.css:183]: All styles other than color and background-color is already there in header nav.nav-middle h5 > a {.
So, i think no need to add here.

File: blocks/header/header.css [hunk 2]
```diff
@@ -170,6 +177,17 @@ header nav.nav-bottom .menu-level-1-item > a {
...
+  color: white;
```
Comment (jindaliiita) [inline blocks/header/header.css:184]: color: var(--white)

**Comment by jindaliiita** [comment]
Few visual differences which i can see

Screenshot 2024-01-29 at 10 47 45 AM
Screenshot 2024-01-29 at 10 48 07 AM
Screenshot 2024-01-29 at 10 48 47 AM

**Comment by RitwikSrivastava** [comment]
@jindaliiita I have made some changes wrt to mobile view and tablet view keeping in the dentistry header consistent with the main foundation site header. 
IN the original dentistry site - in mobile view there is a horizontal scroll bar which we don't have (as it is not recommended and also to keep the header consistent)

File: blocks/header/header.js [hunk 3]
```diff
@@ -35,16 +38,19 @@ function getNavbarToggler() {
...
   navbarToggl.addEventListener('click', () => {
     const navBottom = document.querySelector('.nav-bottom');
+    const navBottomRight = document.querySelector('.nav-bottom-right');
     const header = document.querySelector('header');
     const { body } = document;
     if (navBottom.classList.contains('open')) {
       navBottom.classList.remove('open');
       header.classList.remove('menu-open');
       body.classList.remove('fixed');
+      navBottomRight.classList.remove('fixed');
```
Comment (jindaliiita) [inline blocks/header/header.js:48]: this will in case of normal header

File: blocks/header/header.js [hunk 4]
```diff
@@ -35,16 +38,19 @@ function getNavbarToggler() {
...
     } else {
       navBottom.classList.add('open');
       header.classList.add('menu-open');
       body.classList.add('fixed');
+      navBottomRight.classList.add('fixed');
```
Comment (jindaliiita) [inline blocks/header/header.js:53]: this will in case of normal header

File: blocks/header/header.js [hunk 5]
```diff
@@ -35,16 +38,19 @@ function getNavbarToggler() {
   }
```
Comment (jindaliiita) [inline blocks/header/header.js:38]: Just wondering what is the need of adding the above styling in js.
Not related to your change but still want to know

File: blocks/header/header.css [hunk 6]
```diff
@@ -76,8 +90,22 @@ header nav.nav-middle p {
   margin: 0;
 }
 
-header nav.nav-middle a {
-  display: flex;
+
+header nav.nav-middle h6 {
```
Comment (jindaliiita) [inline blocks/header/header.css:94]: this change will also impact the normal header, is that correct ?
as nav-middle is also present there

File: blocks/header/header.js [hunk 7]
```diff
@@ -119,18 +130,36 @@ export default async function decorate(block) {
...
+    if (folder) {
+      const bottomnav = document.createElement('nav');
+      bottomnav.classList.add('nav-bottom-parent');
+      const nav = document.createElement('nav');
+      nav.classList.add('nav-bottom');
+      navDecorators['nav-bottom'](nav, placeholders, navTreeJson);
+      bottomnav.appendChild(nav);
+      const nav1 = document.createElement('nav');
+      nav1.classList.add('nav-bottom-right');
+      nav1.innerHTML += fetchedNav.querySelectorAll(':scope>div')[idxcounter + 1].innerHTML;
```
[the remark this answers is not in the document]
Comment (RitwikSrivastava) [reply blocks/header/header.js:154]: I divided the nav-bottom into two parts. nav-bottom (existing) + nav-bottom-right which contains the two buttons and phone number.

nav-top.innerHTML is the first section in the dentistrynav.doc. 
nav-middle.innerHTML is the second section. 
nav-bottom-right.innerHTML is the third section from the doc.

File: blocks/header/header.css (hunk 6 above)
Comment (RitwikSrivastava) [reply blocks/header/header.css:97]: nav-middle h6 is not present in normal header doc. So wont be affecting it.

File: blocks/header/header.css (hunk 1 above)
Comment (RitwikSrivastava) [reply blocks/header/header.css:205]: updated.

File: blocks/header/header.js (hunk 5 above)
Comment (RitwikSrivastava) [reply blocks/header/header.js:38]: When the window is downsized, the nav items are displayed inside the mobile-icon toggler. It seems it was added as an placeholder for navbar toggler.

File: blocks/header/header.js [hunk 8]
```diff
@@ -119,18 +130,36 @@ export default async function decorate(block) {
...
+      bottomnav.appendChild(nav1);
+      block.appendChild(bottomnav);
+    } else {
+      const nav = document.createElement('nav');
```
[the remark this answers is not in the document]
Comment (RitwikSrivastava) [reply blocks/header/header.js:162]: We can extract the two lines below (not the "block.appendChild(nav)") , but for segregation for diff headers, I kept them them separately.

**Comment by JiangLong2019** [comment]
Let add few findings.
1. The 2 buttons looks bigger than original site and missing the bottom padding.
Original site:
image

EDS:
image

2. The button and tel number position is different in tablet & mobile view
Original site:
image

EDS:
image

**Comment by RitwikSrivastava** [comment]
@JiangLong2019 It is fixed now. Please review. 
Screenshot 2024-02-07 at 2 02 37 PM

**Comment by JiangLong2019** [comment]
@RitwikSrivastava Looks good to me.

**Comment by jindaliiita** [comment]
I have not checked checked the UI but changes LGTM.

**Review by jindaliiita — APPROVED** [review]

**Pull request merged**